internal/pkg/term/graph/graph.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package graph provides functionality for graphes.
+package graph
+
+type nodeColor int
+
+const (
+	white nodeColor = iota + 1
+	gray
+	black
+)
+
+// Graph represents a directed graph.
+type Graph struct {
+	Nodes map[string][]string
+}
+
+// NewGraph initiates a new Graph.
+func NewGraph() *Graph {
+	return &Graph{
+		Nodes: make(map[string][]string),
+	}
+}
+
+// Add adds a connection between two Nodes.
+func (g *Graph) Add(fromNode, toNode string) {
+	hasNode := false
+	// Add origin node if doesn't exist.
+	if _, ok := g.Nodes[fromNode]; !ok {
+		g.Nodes[fromNode] = []string{}
+	}
+	// Check if edge exists between from and to Nodes.
+	for _, node := range g.Nodes[fromNode] {
+		if node == toNode {
+			hasNode = true
+		}
+	}
```

Comment:
hmm I don't think this is a very good idea to represent a graph since it is not like a tree and has no "root node". To make it more efficient maybe
```golang
type Graph struct {
	Nodes map[string]Neighbor
}

type Neighbor map[string]bool
```
Do you think it is better?